zmon-controller-app/src/main/java/org/zalando/zmon/controller/MultiKairosDBController.java

```diff
@@ -189,10 +189,13 @@ private void alignQueriesToFullMinutes(final JsonNode node, final int queryWindo
                 start = start - (start % 60000);
                 q.put("start_absolute", start);
             } else if (q.has("start_relative")) {
-                if (queryWindow != 0) {
-                    ObjectNode r = (ObjectNode) q.get("start_relative");
-                    r.put("value", queryWindow);
-                    r.put("unit", "minutes");
+                if (maxQueryWindow != 0) {
+                    int curWindow = q.get("start_relative").asInt();
```

Comment:
Looking at the code below, it seems that `start_relative` is an object of keys `value` and `unit`.
This means it becomes a bit more tricky, since you need to convert the range to minutes first in order to be able to compare with the max window setting.